src/incubator/TextField/index.tsx

```diff
@@ -124,18 +126,42 @@ const TextField = (props: InternalTextFieldProps) => {
     return [inputStyle, styles.dummyPlaceholder];
   }, [inputStyle]);
 
-  return (
-    <FieldContext.Provider value={context}>
-      <View {...containerProps} style={[margins, positionStyle, containerStyle, centeredContainerStyle]}>
+  let textFieldLabel = (
+    <Label
+      label={label}
+      labelColor={labelColor}
+      labelStyle={_labelStyle}
+      labelProps={labelProps}
+      floatingPlaceholder={floatingPlaceholder}
+      validationMessagePosition={validationMessagePosition}
+      testID={`${props.testID}.label`}
+    />
+  );
+  const textFieldIsRequired =
+    (typeof others.validate === 'string' && others.validate === 'required') ||
+    (Array.isArray(others.validate) && others.validate.includes('required'));
```

Comment:
Let's move this part of logic to the hook that responsible for the field state (useFieldState)
You can add isMandatory to the fieldState object